### PR TITLE
feat: implemented 'witnet node config' command

### PR DESCRIPTION
**Pull request by twjang**
Hello!

I implemented show-config command. I made the Config struct be able to be converted into PartialConfig, so that it can be serialized into toml format.

The related issue is #1075

### CODE REVIEW

File: config/src/config.rs [hunk 1]
```diff
@@ -303,6 +360,13 @@ pub struct Storage {
     pub master_key_import_path: Option<PathBuf>,
 }
 
+fn to_protected_string<S>(_val: &Option<Protected>, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: serde::Serializer,
+{
+    serializer.serialize_str("**** REDACTED ****")
```
Comment (tmpolaczyk) [inline config/src/config.rs:367]: This should check if the password is `None`, because in that case we must keep it as `None`

Comment (twjang) [reply config/src/config.rs]: I modified this part as you suggested in the latest commit 2a6fc17. Could you review new code? Thanks in advance.

File: config/src/config.rs [hunk 2]
```diff
@@ -245,6 +278,17 @@ pub struct Connections {
     pub reject_sybil_inbounds: bool,
 }
 
+fn to_millis<S>(val: &Option<Duration>, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: serde::Serializer,
+{
+    if let Some(duration) = val {
+        serializer.serialize_u64(u64::try_from(duration.as_millis()).unwrap())
```
[the remark this answers is not in the document]
Comment (twjang) [reply config/src/config.rs]: I changed `serde::de::Error::custom` to `serde::ser::Error::custom` to fix compile error in the latest commit.

**Comment by tmpolaczyk** [comment]
Ok, good! Before merging, can you squash all the commits into one? You can do that by running `git rebase -i upstream/master` and then replace "pick" with "f" for all lines except the first one.

Thank you!

**Review by tmpolaczyk — APPROVED** [review]

**Pull request merged**